### PR TITLE
Jormun: manage places/uri from mimir

### PR DESCRIPTION
**Pull request by azime**
- [x] Tests Bragi Error

### CODE REVIEW

File: source/jormungandr/jormungandr/autocomplete/geocodejson.py [hunk 1]
```diff
@@ -50,18 +50,51 @@ class GeocodeJson(AbstractAutocomplete):
...
+
+        if not self.host:
+            raise TechnicalError('global autocomplete not configured')
+
+        url = "{host}/{end_point}".format(host=self.host, end_point=end_point)
+        if uri:
+            url = '{url}/{uri}'.format(url=url, uri=uri)
+        return url
+
+    def basic_params(self, instance):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/geocodejson.py:98]: I'm not against, but by curiosity why did you change this ?

File: source/jormungandr/jormungandr/autocomplete/geocodejson.py [hunk 2]
```diff
@@ -50,18 +50,51 @@ class GeocodeJson(AbstractAutocomplete):
...
+        try:
+            return method(url, **kwargs)
+        except requests.Timeout:
+            logging.getLogger(__name__).error('autocomplete request timeout')
+            raise TechnicalError('external autocomplete service timeout')
+        except:
+            logging.getLogger(__name__).exception('error in autocomplete request')
+            raise TechnicalError('impossible to access external autocomplete service')
+
+    def response_marshaler(self, response_bragi):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/geocodejson.py]: you can make this either a free function of a static method I think

File: source/jormungandr/jormungandr/autocomplete/geocodejson.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/jormungandr/autocomplete/geocodejson.py:98]: This method should return the common parameters to "get" and "get_uri", common_params ?

File: source/jormungandr/jormungandr/autocomplete/geocodejson.py (hunk 2 above)
Comment (azime) [reply source/jormungandr/jormungandr/autocomplete/geocodejson.py]: ok

File: source/jormungandr/jormungandr/autocomplete/geocodejson.py [hunk 3]
```diff
@@ -50,18 +50,63 @@ class GeocodeJson(AbstractAutocomplete):
...
+            raise TechnicalError('external autocomplete service timeout')
+        except:
+            logging.getLogger(__name__).exception('error in autocomplete request')
+            raise TechnicalError('impossible to access external autocomplete service')
+
+    @classmethod
+    def _check_response(cls, response, uri):
+        if response == None:
+            raise TechnicalError('impossible to access autocomplete service')
+        if response.status_code == 404:
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/autocomplete/geocodejson.py:71]: I don't think it is an invalid argument, it's just that we don't know it :wink: 

aren't there an UnknownObject error (or something like this) ?

Comment (azime) [reply source/jormungandr/jormungandr/autocomplete/geocodejson.py:71]: ok

**Review by antoine-de — APPROVED** [review]
